### PR TITLE
Add track size on font objects

### PR DESCRIPTION
**Pull request by ebraminio**
PTAL

### CODE REVIEW

File: util/options.hh [hunk 1]
```diff
@@ -455,6 +455,7 @@ struct font_options_t : option_group_t
     x_ppem = 0;
     y_ppem = 0;
     ptem = .0;
+    track = .0;
```
Comment (behdad) [inline util/options.hh]: Both of the lines, please use `0.` instead of `.0`.

File: util/options.cc [hunk 2]
```diff
@@ -538,6 +538,7 @@ font_options_t::add_options (option_parser_t *parser)
 			      G_OPTION_ARG_CALLBACK,	(gpointer) &parse_font_size,	font_size_text,					"1/2 integers or 'upem'"},
     {"font-ppem",	0, 0, G_OPTION_ARG_CALLBACK,	(gpointer) &parse_font_ppem,	"Set x,y pixels per EM (default: 0; disabled)",	"1/2 integers"},
     {"font-ptem",	0, 0, G_OPTION_ARG_DOUBLE,	&this->ptem,			"Set font point-size (default: 0; disabled)",	"point-size"},
+    {"font-ptem",	0, 0, G_OPTION_ARG_DOUBLE,	&this->track,			"Set font track-size (default: 0; disabled)",	"track-size"},
```
Comment (behdad) [inline util/options.cc]: Forgot to change option name.

File: docs/harfbuzz-sections.txt [hunk 3]
```diff
@@ -253,6 +253,7 @@ hb_font_get_parent
 hb_font_get_ppem
 hb_font_get_ptem
 hb_font_get_scale
+hb_font_get_track
```
Comment (behdad) [inline docs/harfbuzz-sections.txt]: What's the Apple API like? I like "tracking" better than "track". Should we predefine LOOSE / NORMAL / etc?
I wish we could combine this with letter-spacing as well.

**Comment by ebraminio** [comment]
Now that I am at it, anything else to add also?

File: docs/harfbuzz-sections.txt (hunk 3 above)
Comment (ebraminio) [reply docs/harfbuzz-sections.txt]: > What's the Apple API like?

I don't know and asked a friend who is more familiar and he had no idea also.

Comment (andlabs) [reply docs/harfbuzz-sections.txt]: Hi; person in question here. As far as Cocoa and Core Text goes, I'm pretty sure all the tracking stuff is handled internally; sorry. You can try finding legacy ATS and ATSUI documentation to see how things are handled there (try http://mirror.informatimago.com/next/developer.apple.com/ for legacy docs)...

**Comment by behdad** [comment]
If Apple doesn't expose it in CoreText I'd say we shouldn't add either. Looks like they are just using trak table as a poor man's optical sizing these days.

**Comment by ebraminio** [comment]
If so, [till getting some evidence it is exposed], lets close this and use default (0). `Apple Color Emoji` has two table `trak` like this,

```
<sfntTable_trak version="1.0000" format="0" >
	<trak direction="horizontal" numTrakEntries="2" numSizes="13">
		<trakEntry index="0" value="-1.0000" nameIndex="291" name="[NAME MISSING (291)]" sizeTableOffset="0x0058">
...
		</trakEntry>
		<trakEntry index="1" value="0.0000" nameIndex="290" name="[NAME MISSING (290)]" sizeTableOffset="0x0058">
...
		</trakEntry>
	</trak>
</sfntTable_trak>
```

Is that refers to regular and bold version (does `Apple Color Emoji` that have that even?) or something similar? This one is from San Fransisco Text:

```
<sfntTable_trak version="1.0000" format="0" >
	<trak direction="horizontal" numTrakEntries="3" numSizes="20">
		<trakEntry index="0" value="-1.0000" nameIndex="291" name="[NAME MISSING (291)]" sizeTableOffset="0x007C">
...
		</trakEntry>
		<trakEntry index="1" value="0.0000" nameIndex="290" name="[NAME MISSING (290)]" sizeTableOffset="0x00A4">
...
		</trakEntry>
		<trakEntry index="2" value="2.0000" nameIndex="292" name="[NAME MISSING (292)]" sizeTableOffset="0x00CC">
...
		</trakEntry>
	</trak>
</sfntTable_trak>
```

**Comment by behdad** [comment]
I know CoreText does some compression for fitting lines.  I wonder if the use this.

**Comment by ebraminio** [comment]
Repeating @andlabs comment which is hide above here,

screen shot 2018-02-25 at 12 38 01 pm

**Pull request closed without merging**